packages/docusaurus-migrate/src/index.ts

```diff
@@ -0,0 +1,707 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as fs from 'fs-extra';
+import importFresh from 'import-fresh';
+import chalk from 'chalk';
+import glob from 'glob';
+import Color from 'color';
+
+import {
+  VersionOneConfig,
+  VersionTwoConfig,
+  ClassicPresetEntries,
+  SidebarEntries,
+} from './types';
+import extractMetadata from './frontMatter';
+import migratePage from './transform';
+import sanitizeMD from './sanitizeMD';
+import path from 'path';
+
+const DOCUSAURUS_VERSION = (importFresh('../package.json') as {version: string})
+  .version;
+
+export function walk(dir: string): Array<string> {
+  let results: Array<string> = [];
+  const list = fs.readdirSync(dir);
+  list.forEach((file: string) => {
+    const fullPath = `${dir}/${file}`;
+    const stat = fs.statSync(fullPath);
+    if (stat && stat.isDirectory()) {
+      results = results.concat(walk(fullPath));
+    } else {
+      results.push(fullPath);
+    }
+  });
+  return results;
+}
+
+function sanitizedFileContent(
+  content: string,
+  migrateMDFiles: boolean,
+): string {
+  const extractedData = extractMetadata(content);
+  const extractedMetaData = Object.entries(extractedData.metadata).reduce(
+    (metaData, value) => {
+      return `${metaData}\n${value[0]}: ${
+        value[0] === 'tags' || !!String(value[1]).match(/^(\w| )+$/m)
+          ? value[1]
+          : `"${value[1]}"`
```

Comment:
does it work if the title already has quotes? or does it lead to duplicate quotes?

Reply:
no duplicate quotes. I actually copied code from D1 to use for migration of frontmatter